qubes-rpc/qubes.DesktopNotify

```diff
@@ -232,11 +231,18 @@ def parse_app_icon(vm, untrusted):
 
     return ""
 
+def strip_markup(untrusted):
+    return re.sub(r"<[^>]*>", "", untrusted)
```

Comment:
I would also include the case where there is no trailing `>`.  Furthermore, a client can use HTML entities to sneak bad characters past us.

Reply:
HTML entities shouldn't be interpreted as html.

But then once I again I cannot possibly know how dumb the downstream implementation possibly is. That's why this feels like fighting windmills.

In fact the spec clearly [states](https://people.gnome.org/~mccann/docs/notification-spec/notification-spec-latest.html#markup) that only the following tags can be expected by notification clients:
```
<b> ... </b> 	Bold
<i> ... </i> 	Italic
<u> ... </u> 	Underline
<a href="..."> ... </a> 	Hyperlink
<img src="..." alt="..."/> 	Image
```

Only `<a>` and `<img>` pose security risks, if not viewed inside a dispVM.

It also says:
"A full-blown HTML implementation is not required of this spec, and notifications should never take advantage of tags that are not listed above." --> Probably 90% of notification servers won't fully support html.

Btw here's the `<a>` related code of the notification server Qubes uses by default:
https://github.com/mate-desktop/mate-notification-daemon/pull/170/files

Reply:
> HTML entities shouldn't be interpreted as html.

No, but they can bypass charset restrictions.